runtime/trappist/src/xcm_config.rs

```diff
@@ -245,13 +252,24 @@ pub struct ReserveAssetsFrom<T>(PhantomData<T>);
 impl<T: Get<MultiLocation>> ContainsPair<MultiAsset, MultiLocation> for ReserveAssetsFrom<T> {
 	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
 		let prefix = T::get();
-		log::trace!(target: "xcm::AssetsFrom", "prefix: {:?}, origin: {:?}", prefix, origin);
-		&prefix == origin &&
-			match asset {
-				MultiAsset { id: xcm::latest::AssetId::Concrete(asset_loc), fun: Fungible(_a) } =>
-					matches_prefix(&prefix, asset_loc),
-				_ => false,
-			}
+		log::trace!(target: "xcm::AssetsFrom", "prefix: {:?}, origin: {:?}, asset: {:?}", prefix, origin, asset);
+		&prefix == origin
+	}
+}
+
+pub struct OnlyTeleportNative;
+impl Contains<(MultiLocation, Vec<MultiAsset>)> for OnlyTeleportNative {
+	fn contains(t: &(MultiLocation, Vec<MultiAsset>)) -> bool {
+		if t.1.len() != 1 {
```

Comment:
```suggestion
t.1.iter().any(|asset| {
    log::trace!(target: "xcm::OnlyTeleportNative", "Asset to be teleported: {:?}", asset);
    match asset {
        MultiAsset { id: xcm::latest::AssetId::Concrete(asset_loc), fun: Fungible(_a) } =>
            matches_prefix(&SelfReserve::get(), asset_loc),
        _ => false,
    }
})
```

I'm in personal conflict with this line: Your proposal is much better in terms of performance and cost of execution since it does not iterate. On the other side, i don't like how the `if t.1.len() != 1` reads and also i have a feeling that this will not escalate much if we need to add another scenario to the teleport ( not clear how a new scenario would fit here).

@stiiifff  what do you think?